test/e2e/apps/job.go

```diff
@@ -513,6 +513,124 @@ done`}
 		gomega.Expect(gotIndexes).To(gomega.Equal(wantIndexes), "expected completed indexes %s, but got %s", wantIndexes, gotIndexes)
 	})
 
+	/*
+		Testcase: Ensure that job with successPolicy succeeded when all indexes succeeded
+		Description: Create an indexed job with successPolicy.
+		Verify that job got SuccessCriteriaMet with SuccessPolicy reason and Complete condition
+		once all indexes succeeded.
+	*/
+	ginkgo.It("with successPolicy should succeeded when all indexes succeeded", func(ctx context.Context) {
+		parallelism := int32(1)
+		completions := int32(1)
+		backoffLimit := int32(6) // default value
+
+		ginkgo.By("Creating an indexed job with successPolicy")
+		job := e2ejob.NewTestJob("succeeded", "with-success-policy-all-index-succeeded", v1.RestartPolicyNever, parallelism, completions, nil, backoffLimit)
+		job.Spec.CompletionMode = ptr.To(batchv1.IndexedCompletion)
+		job.Spec.SuccessPolicy = &batchv1.SuccessPolicy{
+			Rules: []batchv1.SuccessPolicyRule{{
+				SucceededCount: ptr.To[int32](1),
+			}},
+		}
+		job, err := e2ejob.CreateJob(ctx, f.ClientSet, f.Namespace.Name, job)
+		framework.ExpectNoError(err, "failed to create job in namespace: %s", f.Namespace.Name)
+
+		ginkgo.By("Awaiting for the job to have the interim SuccessCriteriaMet with SuccessPolicy reason condition")
+		err = e2ejob.WaitForJobCondition(ctx, f.ClientSet, f.Namespace.Name, job.Name, batchv1.JobSuccessCriteriaMet, batchv1.JobReasonSuccessPolicy)
```

Comment:
if I understand this correctly, the test would complete in a same way without the SuccessPolicy. Only the reason would be different here, right?

Reply:
That's right.